File: PracticeRecursion.java
public class PracticeRecursion {
    public static void main(String[] args) {
        greeting(1);
        System.out.print(factorial(10));
    }

    public static void greeting(int n) {
        System.out.println("hello");

        if (n > 0) {
            greeting(n - 1);
        }
    }

    public static long factorial(int n) {
        if (n == 1 || n == 0) {
            return 1;
        } else {
            return n * factorial(n - 1);
        }
    }
}

File: BasicPractice/PracticeRecursion.py
def greeting(n: int):
    print("hello")
    if n > 0:
        greeting(n - 1)


def factorial(n: int):
    if n == 1:
        return 1
    if n == 0:
        return 1
    return n * factorial(n - 1)


if __name__ == "__main__":
    greeting(5)
    print(factorial(5))
